### PR TITLE
R6 objects are documented

### PR DESCRIPTION
**Pull request by r-ash**
This PR will

* Set up documentation of R6 `DataImport` 
* Add custom `help` and `format` functions for `DataImport` object

### CODE REVIEW

File: man/DataImport.Rd [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+\cr\emph{Usage:}\preformatted{transform()}
+}
+\item{\code{load}}{
+Run the load stage of the data import
+\cr\emph{Usage:}\preformatted{load(comment = NULL, dry_run = FALSE, force = FALSE)}
+
+\emph{Arguments:}
+\itemize{
+\item{\code{comment}:   Optional comment which will be persisted in the log of the import run in the database.
+}\preformatted{\item{\code{dry_run}:   Whether to run in dry run mode. If TRUE then any database changes will be rolled back. Defaults to FALSE.
```
Comment (r-ash) [inline man/DataImport.Rd]: This is one thing I wanted to ask about.

Where is this `\preformatted` tag coming from? I cannot work this out at all. The `man-roxygen/DataImport.R` file generated by `autodoc.R` script looks equivalent to the examples in `vaultr` yet when I generate the documentation here using `devtools::document(roclets=c('rd', 'collate', 'namespace'))` it inserts a `preformatted` block after the first arg. Do you have any ideas why this is? I wonder if I am just missing something.

Comment (richfitz) [reply man/DataImport.Rd]: You have `Roxygen: list(markdown = TRUE)` in your DESCRIPTION

File: R/dettl.R [hunk 2]
```diff
@@ -1,3 +1,21 @@
+##' Manage data import.
+##'
+##' @template DataImport
+##'
+##' @title Data Import
+##' @name DataImport
+##'
+##' @examples
+##' path <- dettl:::prepare_test_import(
+##' system.file("examples", "person_information", package = "dettl"),
```
Comment (richfitz) [inline R/dettl.R]: please indent lines 10 and 11

File: man-roxygen/DataImport.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  params:
+    comment: >-
+      Optional comment which will be persisted in the log of the import run
+      in the database.
+    dry_run: >-
+      Whether to run in dry run mode. If TRUE then any database changes will
+      be rolled back. Defaults to FALSE.
+    force: >-
+      If TRUE then checks that repo is up to date with git remote will be
+      skipped. Defautls to FALSE.
```
Comment (richfitz) [inline man-roxygen/DataImport.yml:21]: `Defautls` - take a look at `spelling::spell_check_package()` and see what it turns up.  I find it works best by:

* add `Language: en-GB` to your DESCRIPTION
* put false positives into a file `inst/WORDLIST` (one word per line)

File: R/format.R [hunk 4]
```diff
@@ -0,0 +1,50 @@
+data_import_format <- function(object, brief, name) {
+  if (brief) {
+    return("Data import object")
+  }
+  nms <- setdiff(ls(object), c("format", "clone", "initialize"))
```
Comment (richfitz) [inline R/format.R]: If you add `cloneable = FALSE` to the class definition you won't have to worry about the clone method

File: R/dettl.R [hunk 5]
```diff
@@ -1,3 +1,21 @@
+##' Manage data import.
```
Comment (richfitz) [inline R/dettl.R:1]: I think you could add pointers here to the top-level functions that people might want to use instead, or give some guidance about when they should look here vs at those functions

**Review by richfitz — CHANGES_REQUESTED** [review]
If you update the markdown setting of your DESCRIPTION the preformatted tag will go away, but you'll break the little bit of markdown you currently have there (a few backticks).  Alternatively you could hack at autodoc.R to remove the indentation

File: man-roxygen/DataImport.yml (hunk 3 above)
Comment (r-ash) [reply man-roxygen/DataImport.yml:21]: Thanks, I did not know about this. I've fixed all of the stuff it flagged up.

File: R/format.R (hunk 4 above)
Comment (r-ash) [reply R/format.R]: Removed `"clone"` here.

File: R/dettl.R (hunk 5 above)
Comment (r-ash) [reply R/dettl.R:1]: I've added some notes about when you'd want to run using the object itself. When the PR with top level functions is merged I will add links to this documentation to those functions and expand the note.

File: man/DataImport.Rd (hunk 1 above)
Comment (r-ash) [reply man/DataImport.Rd]: Nice, I've fixed that and used `\code` where I was using back ticks.

**Review by richfitz — APPROVED** [review]

**Pull request merged**